File: Backend/src/models/CartItem.ts
import { Schema, model, Document } from "mongoose";
import { CartDoc } from "./Cart.js";
import { ProductDoc } from "./Product.js";

export type CartItemDoc = {
  cartId: Schema.Types.ObjectId | CartDoc;
  itemId: Schema.Types.ObjectId | ProductDoc;
  itemQty: number;
};
const CartItemSchema = new Schema(
  {
    cartId: {
      type: Schema.Types.ObjectId,
      ref: "Cart",
    },
    itemId: {
      type: Schema.Types.ObjectId,
      ref: "Product",
    },
    itemQty: { type: Number, default: 1 },
  },
  { timestamps: true }
);

export const CartItem = model("CartItem", CartItemSchema);

File: python/Backend/application/models/CartItem.py
from datetime import datetime
from application.models.db import *



class CartItem(Document):
    timestamps = DateTimeField(default=datetime.utcnow())
    cartId=ObjectIdField()
    itemId= ListField(ObjectIdField())
    itemQty=IntField(default=1)
    def to_json(self):
        return {
            "timestamps": self.timestamps,
            "cartId": self.cartId,
            "itemId": self.itemId,
            "itemQty": self.itemQty,
        }